tests/SimplifyTest.elm

```diff
@@ -15448,6 +15449,201 @@ a = Array.empty
+"""
+                        ]
+        , test "should replace Array.length (Array.fromList [b, c, d]) by 3" <|
+            \() ->
+                """module A exposing (..)
+import Array
+a = Array.length (Array.fromList [b, c, d])
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "The length of the array is 3"
+                            , details = [ "The length of the array can be determined by looking at the code." ]
+                            , under = "Array.length"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Array
+a = 3
+"""
+                        ]
+        , test "should replace Array.empty |> Array.length by 0" <|
+            \() ->
+                """module A exposing (..)
+import Array
+a = Array.empty |> Array.length
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "The length of the array is 0"
+                            , details = [ "The length of the array can be determined by looking at the code." ]
+                            , under = "Array.length"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Array
+a = 0
+"""
+                        ]
+        , test "should replace Array.length (Array.repeat 1 x) by 1" <|
+            \() ->
+                """module A exposing (..)
+import Array
+a = Array.length (Array.repeat 1 x)
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "The length of the array is 1"
+                            , details = [ "The length of the array can be determined by looking at the code." ]
+                            , under = "Array.length"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Array
+a = 1
+"""
+                        ]
+        , test "should replace Array.length (Array.repeat 0 x) by 0" <|
+            \() ->
+                """module A exposing (..)
+import Array
+a = Array.length (Array.repeat 0 x)
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "The length of the array is 0"
+                            , details = [ "The length of the array can be determined by looking at the code." ]
+                            , under = "Array.length"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Array
+a = 0
+"""
```

Comment:
This error gets removed in the next commits so that we only have 1 error reported instead of 2.